bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: refactor Makefile moq generation

## Description

This PR provide a Makefile target that allows autogeneration of
moqs without generating any other autogenerated elements like 
openapi files.

This allows users of the Makefile to only regenerate the moq files
if that's what they only need to change, and also reduce the time
needed to generate them if that's the only task they need to perform.

The set of changes also refactor the generate target to use
the new moq generation target.

Sometimes, depending on the changes made to a moq it is desirable
to delete the already existing moq file(s) so moq regenerates them
properly. A PRE_DELETE_AUTOGENERATED_MOQ_FILES Makefile variable
has been provided to allow enabling/disabling pre-deleting the
autogenerated moq files before running the moq generation. It is
enabled by default.

Some additional changes have also been included in this PR:
* A go generate moq specification for AMS quota service interface has been added, which was incorrectly missing
* All moq files have been regenerated from zero. This has been done to make sure a given moq generation
  with a specific moq version will also generate the same content and that we have a "clean" starting point after this PR.

## Verification Steps
* Run `make generate` and observe how both the moq files and openapi files are regenerated
* Run `make moq/generate` and observe how moq files are regenerated and that's the only task performed
* Run `PRE_DELETE_AUTOGENERATED_MOQ_FILES make moq/generate` and observe how moqs are regenerated but no pre-deletion of the moq files is performed
* Tests pass


## Checklist (Definition of Done)
